### PR TITLE
fix(settings): disable Discover when appstore is disabled

### PR DESCRIPTION
**Pull request by ShGKme**
## Summary

Even when the App Store is disabled, the default page (also listed in the navigation) is Discovery, which doesn't work. And when there is no internet, it results in error: "Avertissement appstoreFetcher Could not get apps from the appstore".

Solution — if the app store is disabled:
- Remove "Discovery" from the navigation
- Use the old "Your apps" page as the default

P.S. It is still possible to open any category, including supported or non-existed by direct link and see an error. Might make sense to explicitly define the list of available categories and check before route enter/update.

🏚️ Before | 🏡 After
---|---
image | image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/settings/src/router/routes.ts [hunk 1]
```diff
@@ -2,8 +2,11 @@
  * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
  * SPDX-License-Identifier: AGPL-3.0-or-later
  */
+import { loadState } from '@nextcloud/initial-state'
 import type { RouteConfig } from 'vue-router'
```
Comment (susnux) [inline apps/settings/src/router/routes.ts:6]: Types :up: 

```suggestion
import type { RouteConfig } from 'vue-router'
import { loadState } from '@nextcloud/initial-state'
```

Comment (ShGKme) [reply apps/settings/src/router/routes.ts:6]: ESLint does not complain ._.

Comment (skjnldsv) [reply apps/settings/src/router/routes.ts:6]: We're the eslint

**Review by nickvergessen — APPROVED** [review]

**Review by susnux — APPROVED** [review]

**Pull request merged**